lib/doorkeeper/orm/active_record/application.rb

```diff
@@ -11,14 +11,9 @@ class Application < ActiveRecord::Base
     end
     has_many :authorized_applications, through: :authorized_tokens, source: :application
 
-    def self.column_names_with_table
-      self.column_names.map { |c| "#{table_name}.#{c}" }
-    end
-
     def self.authorized_for(resource_owner)
-      joins(:authorized_applications).
-        where(AccessToken.table_name => { resource_owner_id: resource_owner.id, revoked_at: nil }).
-        group(column_names_with_table.join(','))
+      where(id: Doorkeeper::AccessToken.
+        where(resource_owner_id: resource_owner.id, revoked_at: nil))
     end
```

Comment:
I do not quite understand the question.

Rails is smart and merged argument query (` Doorkeeper::AccessToken.where(resource_owner_id: resource_owner.id, revoked_at: nil)`) to the first query `(where(id: ...))` and make a one SQL request with subquery:

```sql
SELECT * FROM oauth_applications
WHERE oauth_applications.id IN (
  SELECT id FROM oauth_access_tokens
  WHERE resource_owner_id: 1 AND revoked_at IS NOT NULL
)
```

I don't think there is a need in extracting of the `Doorkeeper::AccessToken.where`, but if you want - I can do it.